scipy/stats/_stats_py.py

```diff
@@ -9440,3 +9442,100 @@ def rankdata(a, method='average', *, axis=None, nan_policy='propagate'):
         result[nan_indexes] = np.nan
 
     return result
+
+
+def expectile(a, alpha=0.5, *, dtype=None, weights=None):
+    r"""Compute the expectile.
+
+
+    Expectiles are a generalization of the expectation in the same way as
+    quantiles are a generalization of the median. The expectile at level
+    `alpha = 0.5` is the mean (average). See Notes for more details.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array or object that can be converted to an array.
+    alpha : float
+        The level of the expectile; `alpha=0.5` gives the mean.
+    dtype : dtype, optional
+        Type to which the input arrays are cast before the calculation is
+        performed.
+    weights : array_like, optional
+        The `weights` array must be broadcastable to the same shape as `a`.
+        Default is None, which gives each value a weight of 1.0.
+
+    Returns
+    -------
+    expectile : ndarray
+        See `dtype` parameter above.
+
+    See Also
+    --------
+    numpy.mean : Arithmetic average
+    numpy.quantile : Quantile
+
+    Notes
+    -----
+    The (empirical) expectile at level `alpha` (:math:`\alpha`) of the array
+    `a` (:math:`a_i`) associated to `weights` (:math:`w_i`) is the unique
+    solution :math:`t` of:
+
+    .. math::
+
+        \alpha \sum_{i=1}^n w_i (a_i - t)_+ =
+            (1 - \alpha) \sum_{i=1}^n w_i (t - a_i)_+ \,.
+
+    References
+    ----------
+    .. [1] W. K. Newey and J. L. Powell (1987), "Asymmetric Least Squares
+           Estimation and Testing," Econometrica, 55, 819-847.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.stats import expectile
+    >>> a =[1, 4, 2, -1]
+    >>> expectile(a, alpha=0.5) == np.mean(a)
+    True
+    >>> expectile(a, alpha=0.2)
+    0.42857142857142855
+    >>> expectile(a, alpha=0.8)
+    2.5714285714285716
+
+    """
+    if alpha < 0 or alpha > 1:
+        raise ValueError(
+            "The expectile level alpha must be in the range [0, 1]."
+        )
+    a = np.asarray(a, dtype=dtype)
+
+    if weights is not None:
+        weights = np.asarray(weights, dtype=dtype)
+
+    # This is the empirical equivalent of Eq. (13) with identification
+    # function from Table 9 (omitting a factor of 2) in
+    # Gneiting, T. (2009). "Making and Evaluating Point Forecasts".
+    # Journal of the American Statistical Association, 106, 746 - 762.
+    # https://doi.org/10.48550/arXiv.0912.0902
```

Comment:
Unfortunately, I've not found a good reference for empirical evaluation of expectiles. I think, however, that it is sounder this way: having formulation on the population level instead of only having a formulation on the sample level (given one doesn't get both;-).

Reply:
Ok. Then does it look ok if I change the implementation of the first order condition to that above?
And the last question I have is what do you think of [this](https://github.com/scipy/scipy/pull/17039#discussion_r1007781224)?

Reply:
Yes, you suggestion is even easier to read. I would, however, use `np.average` instead of `np.sum`
```python
return np.average(np.abs((a < t) - alpha) * (a - t), weights=weights)
```